src/Compilers/CSharp/Test/Syntax/Parsing/RecordParsing.cs

```diff
@@ -2608,5 +2608,915 @@ public void Base_05()
             }
             EOF();
         }
+
+        [Fact, CompilerTrait(CompilerFeature.RecordStructs)]
+        public void RecordStructParsing_RecordNamedStruct()
+        {
+            var text = "record struct(int X, int Y);";
+            UsingTree(text, options: TestOptions.Regular9,
+                // (1,8): error CS8652: The feature 'record structs' is currently in Preview and *unsupported*. To use Preview features, use the 'preview' language version.
+                // record struct(int X, int Y);
+                Diagnostic(ErrorCode.ERR_FeatureInPreview, "struct").WithArguments("record structs").WithLocation(1, 8),
+                // (1,14): error CS1001: Identifier expected
+                // record struct(int X, int Y);
+                Diagnostic(ErrorCode.ERR_IdentifierExpected, "(").WithLocation(1, 14)
+                );
+            N(SyntaxKind.CompilationUnit);
```

Comment:
consider extracting the generated bits in the new tests to local functions that are called after each `UsingTree`.